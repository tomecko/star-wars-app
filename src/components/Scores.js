import React from 'react';
import { Card, Progress } from 'reactstrap';

import { getScoresMax } from '../helpers';

export function Scores({ scores }) {
  return (
    <Card body>
      {Object.keys(scores).map(i => (
        <div key={i}>
          Player {Number(i) + 1}
          <Progress value={Math.max(scores[i], 0.5)} max={getScoresMax(scores)}>
            {scores[i] > 0 ? scores[i] : '0'}
          </Progress>
        </div>
      ))}
    </Card>
  );
}

