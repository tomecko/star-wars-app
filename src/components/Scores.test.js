import React from 'react';
import renderer from 'react-test-renderer';

import { Scores } from './Scores';

test('matches snapshot', () => {
  const component = renderer.create(
    <Scores scores={{ 0: 1, 1: 0}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
