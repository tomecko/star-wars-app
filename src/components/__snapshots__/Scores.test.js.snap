// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card card-body"
>
  <div>
    Player 
    1
    <div
      className="progress"
    >
      <div
        aria-valuemax={10}
        aria-valuemin="0"
        aria-valuenow={1}
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "width": "10%",
          }
        }
      >
        1
      </div>
    </div>
  </div>
  <div>
    Player 
    2
    <div
      className="progress"
    >
      <div
        aria-valuemax={10}
        aria-valuemin="0"
        aria-valuenow={0.5}
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "width": "5%",
          }
        }
      >
        0
      </div>
    </div>
  </div>
</div>
`;
